synapse/handlers/auth.py

```diff
@@ -680,13 +680,16 @@ def validate_login(self, username, login_submission):
 
         if login_type == LoginType.PASSWORD:
             known_login_type = True
+            if not self.hs.config.password_localdb:
+                raise LoginError(403, "Local DB Authentication Disabled",
```

Comment:
I'd let this fall through to the "unknown login type"/"Invalid password" code below. I don't think "Local DB Authentication Disabled" is a helpful message for users.

In other words, just change line 681 to be:

```
if login_type == LoginType.PASSWORD and self.hs.config.password_localdb:
```